### PR TITLE
Response level requests do not return response information

### PR DESCRIPTION
**Pull request by nick-iris**
The ph5tostationxml.py needs to include response information for response level request.

I've added some foundational code for this purpose.

### CODE REVIEW

**Comment by derick-hess** [comment]
Thanks for the headstart. I'll get on that tonight or tomorrow morning.

**Comment by nick-iris** [comment]
One potential issue I found with relying on the NRL is that the PH5 Sensor and Datalogger vender, manufacturer, and models will ideally need to match the keys that the NRL Client expects. Otherwise you'll need a method to perform the mapping of PH5 Metadata to NRL Client Keys.

The NRL Client uses the following keys:

```
NRL library at http://ds.iris.edu/NRL
  Sensors: 24 manufacturers
    'CEA-DASE', 'CME', 'Chaparral Physics', 'Eentec', 'GEObit',
    'GEOsig', 'Generic', 'Geo Space/OYO', 'Geodevice', 'Geotech',
    'Guralp', 'Hyperion', 'IESE', 'Kinemetrics', 'LaHusen', 'Lennartz'
    'Metrozet', 'Nanometrics', 'REF TEK', 'Sercel/Mark Products',
    'Silicon Audio', 'SolGeo', 'Sprengnether (now Eentec)',
    'Streckeisen'
  Dataloggers: 15 manufacturers
    'Agecodagis', 'DAQ Systems (NetDAS)', 'Earth Data', 'Eentec',
    'Generic', 'GeoBIT', 'Geodevice', 'Geotech', 'Guralp',
    'Kinemetrics', 'Nanometrics', 'Omnirecs', 'Quanterra', 'REF TEK',
    'SolGeo'
```
Have you been careful to make sure the metadata in each PH5 experiment matches these exactly?

**Comment by derick-hess** [comment]
Yeah, last week the data group here created a spreadsheet to make sure the naming convention was consistent and these are what we are using from now on. I've almost finished updating the experiments already in PN4 to use this naming convention. The only experiment that hasn't been updated is 16-013 and I'll have that fixed tonight

**Comment by nick-iris** [comment]
The NRL Client [get_response](https://github.com/obspy/obspy/blob/RESPtoInventoryResponse/obspy/clients/nrl/client.py#L182) method requires that you provide the sensor manufacturer, model, and natural frequency for the sensor_keys parameter.

i.e.
manufacturer = Geo Space/OYO
model = GS-11D
natural frequency = ???? ('10 Hz', '4.5 Hz')

Do you where the natural frequency of a sensor can be found PH5?

**Comment by nick-iris** [comment]
Besides the sensor and datalogger manufacturer & model, I think we'll need the following additional information to use the NRL Client. I'm not sure if the sensor natural frequency and datalogger family model are in PH5.

-  **sensor natural frequency**
`print(nrl.sensors[Geo Space/OYO][GS-11D])` returns
_Select the natural frequenct for the GS-11D (2 items):
  '10 Hz', '4.5 Hz'_
- **datalogger family type:**
`print(nrl.dataloggers[REF TEK][RT 125 & 125A])` returns:
_Select the RT125 family model (2 items):
  'RT125 - fixed preamp gain', 'RT125A - variable preamp gain'_
- **datalogger sample rate**
`print(nrl.dataloggers[REF TEK][RT 125 & 125A]['RT125 - fixed preamp gain'])` returns:
_Select the sample rate in samples per second (16 items):
  '1', '10', '100', '1000', '125', '2', '20', '200', '25', '250', '4'
  '40', '5', '50', '500', '8'_

**Comment by derick-hess** [comment]
Using the NRL is going to only have to be used for now until we have resp files embedded in the PH5 files. PH5 experiments only use a very small and specific subset of the NRL instruments.  For instance all the GS11-D geophones are 4420 ohm shunt resistance. So I can make it work for now.

The datalogger family field is a bit strange. It is literally only for rt125s in the NRL as far as I can find.

I do think in addition to moving to embedded resp files in PH5 we also should probably add a few more fields into the resp_g to represent all of this extra info.

**Comment by nick-iris** [comment]
I too was originally hoping that the NRL would provide an easy fix, but after looking into this more I've discovered otherwise. The NRL Client will require additional information that is not currently in PH5 to be correctly implemented.

We really shouldn't hardcode the "GS11-D  4420 ohm shunt resistance" etc.. in the API code, and it is going to take work to add all of the required info for the NRL Client into the PH5 experiments. 

I personally think it would be a far better solution to just create RESP files and stored them in PH5 from the beginning rather than messing around with the NRL. Even if this means that we have fewer experiments made available on the initial release of the Web Service.

It looks like the new ObsPy code includes the addition of a obspy.io.xseed feature for reading RESP. See the following commit https://github.com/obspy/obspy/pull/1185/commits/a449a2066a8e579effb3d6bdbfcd2172cd5ad872. We can use this to read the embedded RESP files.

**Comment by nick-iris** [comment]
@rsdeazevedo do you have an opinion on this?

**Comment by nick-iris** [comment]
I also just thought of a potential way to speed up the process of creating the RESP files. Assuming that you know the values for the additional fields that are required for the NRL Client to work, then we could possibly do the following:

1. Temporarily hard code the additional required values for the NRL Client (natural frequency, coil resistance, shunt resistance, etc...)
2. Make a request to the NRL using the additional hardcoded values
3. Use the obspy.io.xseed module to write the response from the NRL to a RESP file.
4. Add the resulting RESP file into the PH5 experiment.

**Comment by derick-hess** [comment]
The more I've been messing with the NRL client, It's just not possible to use it without hard coding a bunch of values which shouldn't be done. I don't think it would be any more work to load the RESP files into PH5 than it would be to write URLs into PH5. I think I can get resp files loaded into  PH5 by monday.

I think  what I will do is load the RESP files into the response_g (a resp for each datalogger and sensor in the experiment) then write those links to the appropriate response_t. From there we can use the obspy xseed parser to load the resp files into an inventory object. 

The only current exception to 2 resp files per datalogger/sensor combo are the nodes. The nodes just use one combined resp file that would be loaded into the link for the datalogger. LLoyd's code now handles this just fine.

**Comment by nick-iris** [comment]
I just pushed a class to manage what RESP files have already been parsed, and to store their response information in memory.

This can be used to keep track of what Sensor/Datalogger RESP pairs have already been parsed, so that we aren't repeatedly parsing the same RESP for multiple channels.

File: kitchen/ph5tostationxml.py [hunk 1]
```diff
@@ -664,6 +679,7 @@ def Process(self):
                         paths.append(dirName)
 
         for path in paths:
+            self.path = path
```
Comment (derick-hess) [inline kitchen/ph5tostationxml.py]: Ahh...this doesn't need to be there. Forgot to remove in while cleaning up.

Comment (nick-iris) [inline kitchen/ph5tostationxml.py]: No worries. I'll remove it in my next commit.

**Comment by nick-iris** [comment]
I updated the code to read the RESP files from PH5 and include them in the StationXML.

If no response files are present or the PH5 experiment hasn't been updated to include the `response_file_das_a` and `response_file_sensor_a` files, then the response info is not included in the output. The code also handles the case where there is only one response file but not both.

**Comment by derick-hess** [comment]
Thanks! I'll test it out in the morning and then get to writing the utility that will add resp files to PH5 experiments so we can get response info into the rest of the experiments.

**Comment by nick-iris** [comment]
Thanks Derick. I'll merge the 2 remaining pull request sometime this morning. It's going to take a bit of finesse since both modified the same files.

**Comment by nick-iris** [comment]
Rebased and merged

**Pull request closed without merging**